Rules/CommonScripts/CinematicCommon.as

```diff
@@ -0,0 +1,303 @@
+{
+	Quicksort(blobs, 0, blobs.length - 1, ComparisonType::x_position);
+}
+
+void SortBlobsByYPosition(CBlob@[]@ blobs)
+{
+	Quicksort(blobs, 0, blobs.length - 1, ComparisonType::y_position);
+}
+
+int partition(CBlob@[]@ arr, int low, int high, u8 comparisonType)
+{
+	CBlob@ pivot = arr[high];
+	int i = low - 1; //index of smaller element
+	for (int j = low; j < high; j++)
+	{
+		bool shouldSwap = false;
+		switch (comparisonType)
+		{
+			case ComparisonType::cinematic_importance:
+				shouldSwap = arr[j].get_f32("cinematic importance") > pivot.get_f32("cinematic importance");
+				break;
+			case ComparisonType::x_position:
+				shouldSwap = arr[j].getPosition().x < pivot.getPosition().x;
+				break;
+			case ComparisonType::y_position:
+				shouldSwap = arr[j].getPosition().y < pivot.getPosition().y;
+				break;
+		}
+
+		if (shouldSwap)
+		{
+			i++;
+
+			//swap arr[i] and arr[j]
+			Swap(arr, i, j);
+		}
+	}
+
+	//swap arr[i+1] and arr[high] (or pivot)
+	Swap(arr, i + 1, high);
+
+	return i + 1;
+}
+
+void Swap(CBlob@[]@ arr, int index1, int index2)
+{
+	CBlob@ temp = arr[index1];
+	@arr[index1] = arr[index2];
+	@arr[index2] = temp;
+}
+
+void Quicksort(CBlob@[]@ arr, int low, int high, u8 comparisonType)
+{
+	if (low < high)
+	{
+		int pi = partition(arr, low, high, comparisonType);
+
+		//recursively sort elements
+		Quicksort(arr, low, pi - 1, comparisonType);
+		Quicksort(arr, pi + 1, high, comparisonType);
+	}
+}
+
+bool focusOnBlob(CBlob@[] blobs)
+{
+	const float CHANGE_FOCUS_DELAY = 1.5f; //time before camera moves on from focused blob
+
+	if (getGameTime() < switchTarget && currentTarget !is null && currentTarget.getHealth() <= 0)
+	{
+		//stay at focus blob's position for a bit after they die
+		posTarget = currentTarget.getInterpolatedPosition();
+		zoomTarget = 1.0f;
+		return true;
+	}
+	else if (blobs.length > 0)
+	{
+		for (uint i = 0; i < blobs.length; i++)
+		{
+			CBlob@ blob = blobs[i];
+			u16 networkID = blob.getNetworkID();
+			@blob = getBlobByNetworkID(networkID);
+
+			if (blob !is null)
+			{
+				if (getGameTime() < switchTarget && blob !is currentTarget)
+				{
+					//stay at focus blob's position for a bit before focusing on a more important blob
+					posTarget = currentTarget.getInterpolatedPosition();
+					zoomTarget = 1.0f;
+				}
+				else
+				{
+					//follow important blob
+					posTarget = blob.getInterpolatedPosition();
+					zoomTarget = 1.0f;
+					@currentTarget = blob;
+					switchTarget = getGameTime() + CHANGE_FOCUS_DELAY * getTicksASecond();
+				}
+
+				return true;
+			}
+		}
+	}
+
+	//not following an important blob
+	return false;
+}
+
+void ViewEntireMap()
+{
+	Vec2f mapDim = getMap().getMapDimensions();
+	posTarget = mapDim / 2.0f;
+	float zoomW = calculateZoomLevelW(mapDim.x);
+	float zoomH = calculateZoomLevelH(mapDim.y);
+	zoomTarget = Maths::Min(zoomW, zoomH);
+	zoomTarget = Maths::Clamp(zoomTarget, 0.5f, 2.0f);
+}
+
+ConfigFile@ openCinematicConfig()
```

Comment:
Once you read the value from the file save it to a blob/rules or global in the script and don't try to read from the file anymore.